### PR TITLE
[otbn] Document key sideload

### LINKED ISSUE #8502: Document/specify feature in OTBN documentation

### PR DESCRIPTION
**Pull request by imphil**
Fixes #8502

### CODE REVIEW

**Review by cdgori — APPROVED** [review]
The error approach seems fine to me.

It's been so long that I don't remember the specific use case we anticipated here, maybe @felixmiller or @tjaychen remembers?

I'm guessing this is to connect an ECDSA p-384 private key for signing.  (Presumably it would also work with p-256, just ignoring the upper bits?)

(But I would think that the case of ECDH to generate a session key would require communication in the other direction, maybe we decided that was out-of-scope here?  The sideload mechanism as currently spec'd doesn't really work like that, I guess.)

**Review by rswarbrick — APPROVED** [review]
Looks good (and nice and simple) to me.

I think we should probably update the block diagram to show the connection though.

File: hw/ip/otbn/doc/_index.md [hunk 1]
```diff
@@ -374,6 +374,46 @@ Reads never stall.
...
       </td>
     </tr>
+    <tr>
+      <td>0x4</td>
+      <td>RO</td>
+      <td>KEY_S0_L</td>
+      <td>
+Bits [255:0] of share 0 of the 384b OTBN sideload key provided by the [Key Manager]({{< relref "/hw/ip/keymgr/doc" >}}).
+
+A `KEY_INVALID` software error is raised if the Key Manager does not provide a valid key.
```
Comment (GregAC) [inline hw/ip/otbn/doc/_index.md:384]:
```suggestion
A `KEY_INVALID` software error is raised on read if the Key Manager has not provided a valid key.
```

Or similar just to make it clear what when the `KEY_INVALID` error is raised.

**Comment by imphil** [comment]
> I think we should probably update the block diagram to show the connection though. 

done.

**Comment by vogelpi** [comment]
I am closing this PR as all this changes have been merged as part of #9350.

**Pull request closed without merging**